proposals/runtime_on_load_on_enabled_events.md

```diff
@@ -0,0 +1,181 @@
+    removeListener(callback: () => void): void;
+  }
+}
+```
+
+#### `chrome.runtime.onExtensionLoaded`
+```typescript
+namespace runtime {
+  // The reason for which the event is being dispatched.
+  //
+  // 'enabled': The extension was re-enabled from a disabled state.
+  //
+  // 'installed': The extension was newly installed.
+  //
+  // 'updated': The extension was reloaded after an update.
+  //
+  // 'profileStart': The extension is being loaded during a profile startup.
+  //
+  // 'reload': The extension was reloaded (e.g. via `chrome.runtime.reload() or`
+  // the user manually reloaded the extension).
+  export type OnLoadedReason = 'enabled' | 
+                               'installed' |
+                               'updated' |
+                               'profileStart' |
+                               'reload';
+
+  export interface ExtensionLoadDetails {
+    // The reason that this event is being dispatched.
+    onLoadReason: OnLoadedReason;
+
+    // Indicates the previous version of the extension, which has just been
+    // updated. This is present only if 'reason' is 'updated' or ‘enabled’.
+    previousVersion?: string;
+  }
+
+  // Fired when the extension is added to the set of active extensions. This can
+  // occur in multiple scenarios: when an extension is first installed, updated
+  // to a new version, and enabled after being disabled.
+  export interface onExtensionLoaded {
+    addListener(callback: (details: ExtensionLoadDetails) => void): void;
+    removeListener(callback: (details: ExtensionLoadDetails) => void): void;
+  }
+}
+```
+
+### Behavior
+
+Described as code comments in schema description.
+
+### New Permissions
+
+No new permissions are needed as extensions can access the `browser.runtime`
+API without any permissions.
+
+### Manifest File Changes
+
+No new manifest changes are added.
+
+## Security and Privacy
+
+### Exposed Sensitive Data
+
+The new events expose some information that wasn’t present before. It does
+expose that an extension was (explicitly) disabled, which is something we don't
+expose today. Today, an extension cannot necessarily distinguish between being
+disabled and the user simply not opening the browser for &lt;n> amount of
+time.
+
+### Abuse Mitigations
+
+These APIs would be difficult to use in an abusive way since they do not give
+information or access to other extensions, and don’t give control into the
+extensions lifecycle – only knowledge as it transitions to different states.
+
+### Additional Security Considerations
+
+N/A.
+
+## Alternatives
+
+### Existing Workarounds
+
+There is currently `chrome.runtime.onInstalled`, and `chrome.runtime.onStartup`.
+These API have gaps in listening that are addressed by these two events. In
+short, `chrome.runtime.onEnabled`, covers the remaining status that developers
+care to know, and `chrome.runtime.onExtensionLoaded` encapsulates them all
+together with a single listener.
```

Comment:
Should the other events be soft deprecated now and slated for removal in the next manifest version bump? (I think so, as they are redundant now.)

Reply:
The impression I got from past discussions is that we see some value in having discrete events, for the cases where you do only want to listen to a particular part of the lifecycle. While I agree they are technically redundant, since we already have them I'm not sure if we should go out of our way to remove them.

Reply:
I agree that the old events are now technically redundant. If we want to soft-deprecate them, I think we can discuss that separately. What do you think?